framework/Optimizers/Conjugate.py

```diff
@@ -0,0 +1,421 @@
+      pass # don't have a history on the first pass
+    self.counter['gradientHistory'][traj][0] = gradient
+
+    return gradient
+
+  def _newOptPointAdd(self, gradient, traj):
+    """
+      This local method add a new opt point based on the gradient
+      @ In, gradient, dict, dictionary containing the gradient
+      @ In, traj, int, trajectory
+      @ Out, varKPlus, dict, new point that has been queued (or None if no new points should be run for this traj)
+    """
+    stepSize = self.counter['lastStepSize'][traj]
+    self.optVarsHist[traj][self.counter['varsUpdate'][traj]] = {}
+    varK = dict((var,self.counter['recentOptHist'][traj][0][var]) for var in self.getOptVars())
+    varKPlus,modded = self._generateVarsUpdateConstrained(traj, stepSize, gradient, varK)
+    if len(self.optTrajLive) > 1 and self.counter['solutionUpdate'][traj] > 0:
+      removed = self._removeRedundantTraj(traj, varKPlus)
+    else:
+      removed = False
+    if removed:
+      return None
+    #if the new point was modified by the constraint, reset the step size
+    if modded:
+      self.counter['lastStepSize'][traj] = self.paramDict['initialStepSize']
+      self.raiseADebug('Resetting step size for trajectory',traj,'due to hitting constraints')
+    self.queueUpOptPointRuns(traj,varKPlus)
+    self.optVarsHist[traj][self.counter['varsUpdate'][traj]] = varKPlus
+    return varKPlus
+
+  def _createPerturbationPoints(self, traj, optPoint, submit=True):
+    """
+      Creates perturbation points based on a provided NORMALIZED data point
+      @ In, traj, int, integer label for current trajectory
+      @ In, optPoint, dict, current optimal point near which to calculate gradient
+      @ In, submit, bool, optional, if True then submit perturbation points to queue
+      @ Out, points, list(dict), perturbation points
+    """
+    points = []
+    distance = self.paramDict['pertDist'] * max(self.counter['lastStepSize'][traj],self.paramDict['initialStepSize'])
+
+    for i in self.perturbationIndices:
+      direction = self._getPerturbationDirection(i, step = self.counter['varsUpdate'][traj])
+      point = {}
+      index = 0
+      for var in self.getOptVars():
+        size = np.prod(self.variableShapes[var])
+        if size > 1:
+          new = np.zeros(size)
+          for v, origVal in enumerate(optPoint[var]):
+            new[v] = origVal + distance*direction[index]
+            new[v] = self._checkBoundariesAndModify(1.0, 0.0, 1.0, new[v], 0.9999, 0.0001)
+            index += 1
+          point[var] = new
+        else:
+          val = optPoint[var] + distance*direction[index]
+          val = self._checkBoundariesAndModify(1.0, 0.0, 1.0, val, 0.9999, 0.0001)
+          index += 1
+          point[var] = val
+      points.append(point)
+
+      if submit:
+        prefix = self._createEvaluationIdentifier(traj,self.counter['varsUpdate'][traj],i)
+        self.submissionQueue[traj].append({'inputs':point, 'prefix':prefix})
+    return points
```

Comment:
question above

Reply:
Changing perturbation points' distance to be the max of line search results or the initial step size, this could avoid the perturbed points getting too close.